app/config/services.yml

```diff
@@ -1,83 +1,121 @@
+    # Indeed, registering new classes whose filename matches a glob pattern as
+    # services like in the two previous sections only works for namespaced classes.
     app.twig.intl_extension:
-        public: false
         class:  Twig_Extensions_Extension_Intl
-        tags:
-            - { name: twig.extension }
 
-    # Defining a form type as a service is only required when the form type
-    # needs to use some other services, such as the entity manager.
-    # See http://symfony.com/doc/current/best_practices/forms.html
-    app.form.type.tagsinput:
-        class: AppBundle\Form\Type\TagsInputType
-        arguments: ['@doctrine.orm.entity_manager']
-        tags:
-            - { name: form.type }
+    # Some classes cannot be fully autowired because their methods accept either
+    # some scalar arguments that Symfony cannot guess or a typehinted dependency
+    # for which the container has at least two registered services matching the
+    # type.
+    #
+    # Both the "AppBundle\Twig\AppExtension" and "AppBundle\EventListener\RedirectToPreferredLocaleSubscriber"
+    # classes have a "__construct()" method that receives a "$locales" scalar argument
+    # that Symfony cannot guess. This is why we must manually and explicitly provide
+    # the wiring of this argument to complete their service definitions.
+    #
+    # To do so, the remaining unwired named arguments must be defined with their
+    # corresponding values. Here, the "$locales" named argument is configured to
+    # receive the value of the global "%app_locales%" parameter defined under the
+    # "parameters" section of the "app/config/config.yml" file.
+    #
+    # Note that the order in which the named arguments are defined below doesn't
+    # matter as they're referenced here by their real names in the PHP code.
+    # Symfony is then smart enough to make the corresponding matching when compiling,
+    # optimizing and dumping the container as a raw PHP class in the cache directory.
+    AppBundle\Twig\AppExtension:
+        $locales: '%app_locales%'
+
+    AppBundle\EventListener\RedirectToPreferredLocaleSubscriber:
+        $locales: '%app_locales%'
 
     # Event Listeners are classes that listen to one or more specific events.
     # Those events are defined in the tags added to the service definition.
     # See http://symfony.com/doc/current/event_dispatcher.html#creating-an-event-listener
-    app.redirect_to_preferred_locale_listener:
-        class: AppBundle\EventListener\RedirectToPreferredLocaleListener
-        arguments: ['@router', '%app_locales%', '%locale%']
-        tags:
-            - { name: kernel.event_listener, event: kernel.request, method: onKernelRequest }
-
-    app.comment_notification:
-        class: AppBundle\EventListener\CommentNotificationListener
+    AppBundle\EventListener\CommentNotificationListener:
         arguments: ['@mailer', '@router', '@translator', '%app.notifications.email_sender%']
```

Comment:
It can be autowired if we simplify the `AppBundle/` section by allowing registering all classes under the `EventListener/` directory. However we would still have to manually wire the last argument that is scalar and cannot be guessed by Symfony.